From 5280511ffc86fb358912e03512b320c29d2d4f66 Mon Sep 17 00:00:00 2001
From: David Mandelberg <david@mandelberg.org>
Date: Fri, 4 Jan 2019 21:17:34 -0500
Subject: [PATCH 1/3] Fix completion after a command-line flag.

Before:
  yadm checkout -f <TAB>  # Completes filenames.
  yadm checkout --yadm-dir <TAB>  # Completes filenames.

After:
  yadm checkout -f <TAB>  # Completes branch names.
  yadm checkout --yadm-dir <TAB>  # Completes filenames.
---
 completion/yadm.bash_completion | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/completion/yadm.bash_completion b/completion/yadm.bash_completion
index 1091e55..34f199f 100644
--- a/completion/yadm.bash_completion
+++ b/completion/yadm.bash_completion
@@ -60,15 +60,17 @@ if declare -F _git > /dev/null; then
 		  ;;
     esac
 
+    local yadm_switches=( $(yadm introspect switches 2>/dev/null) )
+
     # this condition is so files are completed properly for --yadm-xxx options
-    if [[ ! "$penultimate" =~ ^- ]]; then
+    if [[ " ${yadm_switches[*]} " != *" $penultimate "* ]]; then
       # TODO: somehow solve the problem with [--yadm-xxx option] being
       #       incompatible with what git expects, namely [--arg=option]
       _git
     fi
     if [[ "$current" =~ ^- ]]; then
       local matching
-      matching=$(compgen -W "$(yadm introspect switches 2>/dev/null)" -- "$current")
+      matching=$(compgen -W "${yadm_switches[*]}" -- "$current")
       __gitcompappend "$matching"
     fi
 
-- 
2.20.1

